client-datastore-common-test/src/main/java/com/telecominfraproject/wlan/client/datastore/BaseClientDatastoreTest.java

```diff
@@ -973,6 +973,63 @@ public void testClientSessionPagination()
         return new ArrayList<>(Arrays.asList(new String[]{"qr_49", "qr_48", "qr_47", "qr_46", "qr_45", "qr_44", "qr_43", "qr_42", "qr_41", "qr_40" }));
     }
     
+    @Test
+    public void testClientSessionPaginationMaxItems()
+    {
+       //create 100 Client sessions
+       ClientSession mdl;
+       int customerId = (int) testSequence.incrementAndGet();
+       long locationId = testSequence.incrementAndGet();
+       
+       int apNameIdx = 0;
+       
+       List<ClientSession> sessionsToCreate = new ArrayList<>();
+       
+       for(int i = 0; i< 10; i++){
+           mdl = new ClientSession();
+           mdl.setCustomerId(customerId);
+           mdl.setLocationId(locationId);
+           mdl.setEquipmentId(testSequence.incrementAndGet());
+           mdl.setMacAddress(new MacAddress("A1:FF:FF:FF:FF:FF"));
+           
+           ClientSessionDetails details = new ClientSessionDetails();
+           details.setApFingerprint("qr_"+apNameIdx);
+           mdl.setDetails(details );
+
+           apNameIdx++;
+           
+           sessionsToCreate.add(mdl);
+       }
+       
+       List<ClientSession> createdList = testInterface.updateSessions(sessionsToCreate);
+       List<ClientSessionDetails> createdDetailsList = new ArrayList<>();
+       List<ClientSessionDetails> detailsToCreateList = new ArrayList<>();
+       
+       sessionsToCreate.forEach(s -> detailsToCreateList.add(s.getDetails()));
+       createdList.forEach(s -> createdDetailsList.add(s.getDetails()));       
+       assertEquals(detailsToCreateList, createdDetailsList);
+       
+       PaginationContext<ClientSession> context = new PaginationContext<>(5);
+       PaginationResponse<ClientSession> page1 = testInterface.getSessionsForCustomer(customerId, null, new HashSet<Long>(Arrays.asList(locationId)), null, Collections.emptyList(), context);
+
+       assertEquals(5, page1.getItems().size());
```

Comment:
@dmitry-toptygin-connectus-ai to respond to your comment from above, I confirmed that the unit test fails here with the old code. The context indicates 5 items to be returned, and the assert line here finds that 10 is returned, and fails. 
This scenario with locationid filter should be the same as the macSubstring filter below, as the client sessions are only differentiated by equipmentId but have the same macAddress (effectively, the client sessions have a unique equipmentId and macAddress pair), so the getSessions from before will get all sessions with matching customerId and macAddress.